frontend/ui/src/pages/Channels/ConnectedChannelsList/ChannelsList.tsx

```diff
@@ -0,0 +1,131 @@
+import React, {useEffect, useState} from 'react';
+import _, {connect, ConnectedProps} from 'react-redux';
+import {withRouter, RouteComponentProps, Link} from 'react-router-dom';
+import {CHANNELS_FACEBOOK_ROUTE, CHANNELS_CHAT_PLUGIN_ROUTE, CHANNELS_ROUTE} from './../../../routes/routes';
+import {ReactComponent as BackIcon} from 'assets/images/icons/arrow-left-2.svg';
+import {ReactComponent as SearchIcon} from 'assets/images/icons/search.svg';
+import {ReactComponent as AddChannelIcon} from 'assets/images/icons/plus.svg';
+import {ReactComponent as CloseIcon} from 'assets/images/icons/close.svg';
+import {StateModel} from './../../../reducers';
+import {Channel} from 'httpclient';
+import {allChannels} from './../../../selectors/channels';
+import {ChannelSource} from 'httpclient';
+import ChannelListItem from './ChannelListItem';
+import {SearchField} from '@airyhq/components';
+import styles from './ChannelsList.module.scss';
+
+type ChannelsListProps = {} & ConnectedProps<typeof connector> & RouteComponentProps<{source: string}>;
+
+const mapStateToProps = (state: StateModel) => ({
+  channels: Object.values(allChannels(state)),
+});
+
+const connector = connect(mapStateToProps, null);
+
+const ChannelsList = (props: ChannelsListProps) => {
+  const {channels} = props;
+  const [name, setName] = useState('');
+  const [path, setPath] = useState('');
+  const [searchText, setSearchText] = useState('');
+  const [showingSearchField, setShowingSearchField] = useState(false);
+  const source = props.match.params.source;
+
+  const filteredChannels = channels
+    .filter((channel: Channel) => channel?.source === source)
+    .filter((channel: Channel) => channel?.metadata?.name?.toLowerCase().includes(searchText.toLowerCase()));
```

Comment:
You can extract this filter code to the `mapStateToProps` to clean up the component code a bit.

Also, I think you can remove the `?` in `channel?.source`, since all channels in that list are present.

Reply:
I just wrote the same in a different place before reading this comment. Yes 100%